### PR TITLE
Minor string-related and style fixes

### PR DESCRIPTION
**Pull request by Stat1cV01D**
I was not able to test the changes, so proceed with care.

### CODE REVIEW

File: gm_dotnet_native/dotnethelper-src/dotnethelper.cpp [hunk 1]
```diff
@@ -36,40 +41,37 @@ typedef cleanup_function_fn(*managed_main_fn)(
...
+using tstring = std::basic_string<char_t>;
+
 std::ofstream error_log_file;
 
 managed_delegate_executor_fn managed_delegate_executor = nullptr;
 
 managed_main_fn managed_main = nullptr;
 
-std::string hosfxr_path = "garrysmod/lua/bin/dotnet/host/fxr/" + std::string(NET_CORE_VERSION);
+const std::filesystem::path bin_folder = _T("garrysmod/lua/bin");
```
Comment (GlebChili) [inline gm_dotnet_native/dotnethelper-src/dotnethelper.cpp]: `bin_folder` var name here can be little bit confusing: Garry's Mod has two bin folders, one for Source Engine binaries and one for Lua native plugins, which we are taking about. I would rather rename it to `lua_bin_folder`.

File: gm_dotnet_native/src/gm_dotnet.cpp [hunk 2]
```diff
@@ -18,9 +19,11 @@ cleanup_function_fn cleanup_function = nullptr;
 //Invoked by Garry's Mod on module load
 GMOD_MODULE_OPEN()
 {
+    const std::filesystem::path bin_folder("garrysmod/lua/bin");
```
Comment (GlebChili) [inline gm_dotnet_native/src/gm_dotnet.cpp]: Same as above, `lua_bin_folder` is a better name for a constant here.

**Review by GlebChili — CHANGES_REQUESTED** [review]
Everything is LGTM, except two constant names.

File: gm_dotnet_native/dotnethelper-src/dotnethelper.cpp (hunk 1 above)
Comment (Stat1cV01D) [reply gm_dotnet_native/dotnethelper-src/dotnethelper.cpp]: I made the changes as you suggested. 

Other than that I would probably propose moving library loading code into a shared header.

**Review by GlebChili — APPROVED** [review]

**Pull request merged**